cpp/oneapi/dal/backend/primitives/reduction/test/reduction_dpc.cpp

```diff
@@ -272,6 +278,57 @@ class reduction_test_random : public te::float_algo_fixture<std::tuple_element_t
     bool override_init_;
 };
 
+template <typename Param>
+class infinite_sum_test_random : public reduction_test_random<Param> {
+public:
+    using float_t = std::tuple_element_t<0, Param>;
+    using binary_t = std::tuple_element_t<1, Param>;
+    using unary_t = std::tuple_element_t<2, Param>;
+
+    void generate(bool maxval) {
+        this->height_ = GENERATE(17, 999, 1, 5, 1001);
+        this->width_ = GENERATE(7, 707, 1, 251, 5);
+        this->override_init_ = GENERATE(0, 1);
+        CAPTURE(this->override_init_, this->width_, this->height_);
+        generate_input(maxval);
+        this->generate_offset();
+    }
+
+    void generate_input(bool maxval) {
+        float_t inp = 0.9 * (float_t)maxval * std::numeric_limits<float_t>::max() + 4.0;
+        const auto train_dataframe = GENERATE_DATAFRAME(
+            te::dataframe_builder{ this->height_, this->width_ }.fill_uniform(-3.0, inp));
+        this->input_table_ = train_dataframe.get_table(this->get_homogen_table_id());
+    }
+};
+
+template <typename Param>
+class single_infinite_test_random : public reduction_test_random<Param> {
+public:
+    using float_t = std::tuple_element_t<0, Param>;
+    using binary_t = std::tuple_element_t<1, Param>;
+    using unary_t = std::tuple_element_t<2, Param>;
+
+    void generate(bool infval) {
+        this->height_ = GENERATE(17, 999, 1, 5, 1001);
+        this->width_ = GENERATE(7, 707, 1, 251, 5);
+        this->override_init_ = GENERATE(0, 1);
+        CAPTURE(this->override_init_, this->width_, this->height_);
+        generate_input(infval);
+        this->generate_offset();
+    }
+
+    void generate_input(bool infval) {
+        const auto train_dataframe = GENERATE_DATAFRAME(
+            te::dataframe_builder{ this->height_, this->width_ }.fill_uniform(-3.0, 4.0));
+        auto inner_iter_count_arr_host = train_dataframe.get_array().get_mutable_data();
```

Comment:
the same comment about naming

Reply:
will definitely change name thanks for the suggestion!